### PR TITLE
Update dodo_ethereum_trades & add dodo_bnb_trades & split them to aggregator and pools

### PR DESCRIPTION
**Pull request by owen05**
Brief comments on the purpose of your changes:

1. Update the original dodo_ethereum_trades
2. Add new dodo_route_proxy
3. Fix dodo_v1 pools' address
4. Add dodo_bnb_trades
5. Split dodo_trades to dodo_aggregator & dodo_pools.

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [x] `coin_id` represents the ID of the coin on coinpaprika.com
* [x] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by owen05** [comment]
@augustog Hey, anything need to fix about this code?

**Comment by jeff-dude** [comment]
@Hosuke do you mind looking into this one, when you have the chance?

**Comment by Hosuke** [comment]
> @Hosuke do you mind looking into this one, when you have the chance?

Sure, I have replied the author in the discord channel.
Since this may be a model merging into dex.aggregator.trades(which is a rare case),
I suggest that, we keep the original dodo trades for dex.trades, and the new dodo models at the same time.
We may remove the deprecated/inaccurate model in the future PR.

**Comment by Hosuke** [comment]
I have discussed with @owen05 , and we agreed that keep both old and new models for now.

**Comment by Hosuke** [comment]
Hi @owen05 , It seems we need to either fix the seeds in the seed file or we restore the DODO_ethereum model, as there are unmatched seeds based on [dbt-test info](https://github.com/duneanalytics/spellbook/actions/runs/3702755046/jobs/6273388717#step:12:45).

File: seeds/dex/trades/dex_trades_seed.csv [hunk 1]
```diff
@@ -55,6 +55,9 @@ ethereum,DODO,2,2022-07-24,0x3d01d3959d15fef7be9876f578f87c0c654e2efafb2d5514211
...
 ethereum,DODO,2,2021-08-26,0xd534643d2f49ee7c7d0020a4e17e9b9e8d7f921e110779fa5c5bd1865e27cb83,290,0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48,100000,0x6b175474e89094c44da98b954eedeac495271d0f,99930.038478836639848083
 ethereum,DODO,2,2021-05-14,0xb386cb8c8412be350f24c05360e29f3319a903363e4d34076352bb35b44bc758,62,0xdac17f958d2ee523a2206206994597c13d831ec7,75000,0x57ab1ec28d129707052df4df418d58a2d46d5f51,74754.450749991178425833
+bnb,DODO,0,2021-12-02,0x3d45d93b6359f62a1ec8134f6d8d1bb130c1ead01d8db57244041e24068b1f85,1926,0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c,0.005982,0xe9e7cea3dedca5984780bafc599bd69add087d56,3.729322856407376809
+bnb,DODO,1,2022-05-09,0x3700c79d69a892d2242a66e5975892fbb623fbcdd58cb05d16526356921b877e,42,0xe9e7cea3dedca5984780bafc599bd69add087d56,999.735607882091781117,0x55d398326f99059ff775485246999027b3197955,1000
+bnb,DODO,2_dvm,2022-06-04,0x6e1d453d23a634d5761cee9ec89d829124e2ad79ddb99b91a861aaed5d2344db,134,0x67ee3cb086f8a16f34bee3ca72fad36f7db929e2,2920.79,0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c,1.670916863658532791
```
Comment (Hosuke) [inline seeds/dex/trades/dex_trades_seed.csv:60]: In my last commits I added 3 seeds for DODO_bnb_trades.
According to the model: version `0` is from `aggregator_trades` while [`1`, `2_dvm`] is from `pool_trades`.

BSCSCAN corresponding to the seeds:
- https://bscscan.com/tx/0x3d45d93b6359f62a1ec8134f6d8d1bb130c1ead01d8db57244041e24068b1f85
- https://bscscan.com/tx/0x3700c79d69a892d2242a66e5975892fbb623fbcdd58cb05d16526356921b877e
- https://bscscan.com/tx/0x6e1d453d23a634d5761cee9ec89d829124e2ad79ddb99b91a861aaed5d2344db

**Comment by owen05** [comment]
@Hosuke Hey, I update seeds about dodo on Ethereum and BNB which were splited by version  field (0-dodo_aggregator, 1-dodoV1, 2_dvm - DVM of dodoV2, 2_dsp - DSP of dodoV2). And about version=0, involving two proxies(DODOV2Proxy & DODORouteProxy), about version=1, involving two trade side(sell basetoken or buy basetoken). Please help a review

File: models/dodo/ethereum/dodo_pools_ethereum_trades.sql [hunk 2]
```diff
@@ -19,12 +19,10 @@
...
-"0xd0678ec59d1337e29d1e1b1b6338f3e85d311f5e", 
-"0x6ed3fd491f5b10384787b3fb162a72227fafbf63",
-"0x9a9942458754bdf65dcbcd0b6b4b842a7d4031ab",
-"0xa2cb66ebb947d217f61510882096f6e95c1de97d",
-"0xa356867fdcea8e71aeaf87805808803806231fdc", 
-"0xa2398842f37465f89540430bdc00219fa9e4d28a"
+"0x91E1c84BA8786B1FaE2570202F0126C0b88F6Ec7",
+"0x9B64c81ba54eA51e1f6B7fefb3cfF8AA6F1e2A09",
+"0xe6AafA1c45D9d0C64686c1f1D17B9fe9c7DAB05b",
+"0xE55154D09265b18aC7CDAC6E646672A5460389a1"
```
Comment (Hosuke) [inline models/dodo/ethereum/dodo_pools_ethereum_trades.sql:25]: We may need to use lowercased addresses here and in other files.

File: seeds/dex/trades/dex_trades_seed.csv [hunk 3]
```diff
@@ -47,14 +47,18 @@ arbitrum,zigzag,1,2022-10-26,0xc36027e8ccac981439cae6471824c08fad2831a94d1824505
...
-ethereum,DODO,2,2021-05-14,0xb386cb8c8412be350f24c05360e29f3319a903363e4d34076352bb35b44bc758,62,0xdac17f958d2ee523a2206206994597c13d831ec7,75000,0x57ab1ec28d129707052df4df418d58a2d46d5f51,74754.450749991178425833
+ethereum,DODO,2_dvm,2022-11-01,0xd22f88ac18cec7940d5d238140e8cd70e1fb77f01524e92cdba9002831405e53,48,0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48,823.892847,0x4bdcb66b968060d9390c1d12bd29734496205581,15050.127244362413
+ethereum,DODO,2_dsp,2022-11-01,0x217a50ce8c056b6f240ca0fee82a7aea76fde5845238717d4d2fe36a1d885fe3,18,0x6b175474e89094c44da98b954eedeac495271d0f,16786.79842759679,0xdac17f958d2ee523a2206206994597c13d831ec7,16789.191028
+bnb,DODO,0,2022-10-27,0x14b34175b6f62736909e0ce38149522f17e1549a9b1502c439921f69ce51236b,129,0x26734add0650719ea29087fe5cc0aab81b4f237d,21053,0x55d398326f99059ff775485246999027b3197955,473.56624201412774
+bnb,DODO,0,2022-10-27,0x89d627300b894d58f00467077f9d040465cf2b3081dbe6336f693e9bf1e358e8,195,0x55d398326f99059ff775485246999027b3197955,498,0x30f1183dec479387c09eab6dab3db88cffad5366,1630.1086595819775
```
Comment (Hosuke) [inline seeds/dex/trades/dex_trades_seed.csv:57]: We may have unmatched seed records here as shown in [dbt-test info](https://github.com/duneanalytics/spellbook/actions/runs/3723044188/jobs/6314313730#step:12:148).

File: models/dodo/ethereum/dodo_pools_ethereum_trades.sql (hunk 2 above)
Comment (owen05) [reply models/dodo/ethereum/dodo_pools_ethereum_trades.sql]: Fixed

File: seeds/dex/trades/dex_trades_seed.csv (hunk 3 above)
Comment (owen05) [reply seeds/dex/trades/dex_trades_seed.csv:57]: update dodo_bnb_schema.yml. version=0 should be contained in dodo_aggregator_bnb_trades

Comment (Hosuke) [reply seeds/dex/trades/dex_trades_seed.csv:57]: You may need to modify the `version` in the seed file as well.

File: models/dodo/bnb/dodo_pools_bnb_trades.sql [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+)
+}}
+    
+{% set project_start_date = '2021-02-22' %}
+
+{% set dodo_proxies = [
+"0x8E4842d0570c85Ba3805A9508Dce7C6A458359d0", 
+"0x0596908263Ef2724fBfBcAfA1c983FCD7a629038",
+"0x165BA87e882208100672b6C56f477eE42502c820",
+"0xAB623fBcaeb522046185051911209F5b2C2a2e1f"
```
Comment (Hosuke) [inline models/dodo/bnb/dodo_pools_bnb_trades.sql:23]: We may need to change these addresses to lowercased.
```suggestion
"0x8e4842d0570c85ba3805a9508dce7c6a458359d0", 
"0x0596908263ef2724fbfbcafa1c983fcd7a629038",
"0x165ba87e882208100672b6c56f477ee42502c820",
"0xab623fbcaeb522046185051911209f5b2c2a2e1f"
```

**Review by Hosuke — APPROVED** [review]
This PR updates dodo_ethereum models and add dodo_bnb models, I have reviewed the spell and here is the summary:

- Metadata & naming conventions look good ✅
- Config blocks match uniqueness tests in schema files ✅
- Tests and seeds added for the new models ✅
- Incremental filters added when joining large tables ✅
- Contract addresses are lowercased ✅
- Inner join to large tables as possible ✅

Moreover for DODO trades, any record with `version 0` indicates it is an aggregator trade, and other versions are for non-aggregator trade.

File: models/dodo/ethereum/dodo_ethereum_schema.yml [hunk 5]
```diff
@@ -96,4 +108,98 @@ models:
...
+            - blockchain
+            - project
+            - version
+            - tx_hash
+            - evt_index
+            - trace_address
+      - check_dex_seed:
+          blockchain: ethereum
+          project: DODO
+          version: 0
```
Comment (Hosuke) [inline models/dodo/ethereum/dodo_ethereum_schema.yml:181]: @owen05 I added an extra test here, as by definition, records with version 0 is for aggregator trades. You can check if the seeds in the seed file match the expectation.

Comment (owen05) [reply models/dodo/ethereum/dodo_ethereum_schema.yml:181]: Thanks. The seeds match the expectation

**Comment by Hosuke** [comment]
> @Hosuke do you mind looking into this one, when you have the chance?

Hi @jeff-dude, this PR is ready for the final review.
Also cc. @augustog

**Comment by Hosuke** [comment]
Moreover the models in this PR may overwrite models in https://github.com/duneanalytics/spellbook/pull/2157.
@LeviScoffie do you want to discuss with @owen05(as you may know each other) which model to keep for the `dodo_bnb_trades`?

**Comment by owen05** [comment]
And I am from the DODO Official Team to restructure our spells on Dune. We plan to launch new features in the future. Before that. Hoping to update currently spells. By now @Hosuke  have already approved this pr.  So can we move to the merging process?  Much appreciated! @jeff-dude

**Comment by jeff-dude** [comment]
@owen05 thank you for the patience through the holidays & code freeze. i've merged main & resolved conflicts after merging first attempt from @LeviScoffie on #2157 

can you take another look and see how it looks?

**Comment by owen05** [comment]
@jeff-dude Thanks for your review. And I update the seeds in dexs.trade about dodo_bnb. So far, everything goes well.

**Review by jeff-dude — APPROVED** [review]
thank you @owen05 -- the final commit was some minor formatting & updated the macro inputs for exposing all the dodo spells. now they will all fall under `dodo` in dune data explorer, with each schema/table within

**Pull request merged**